### PR TITLE
QueryBuilderGetQueryDynamicReturnTypeExtension: handle regular set usages

### PR DESCRIPTION
**Pull request by janedbal**

### CODE REVIEW

File: src/Type/Doctrine/QueryBuilder/QueryBuilderGetQueryDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -152,6 +153,18 @@ public function getTypeFromMethodCall(
...
 				}
 
+				if ($lowerMethodName === 'set') {
+					try {
+						$args = $this->argumentsProcessor->processArgs($scope, $methodName, array_slice($calledMethodCall->getArgs(), 0, 1));
+					} catch (DynamicQueryBuilderArgumentException $e) {
+						return $defaultReturnType;
+					}
+					if (count($args) === 1) {
+						$queryBuilder->set($args[0], $args[0]);
```
Comment (ondrejmirtes) [inline src/Type/Doctrine/QueryBuilder/QueryBuilderGetQueryDynamicReturnTypeExtension.php:163]: This branch does not seem to be tested

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Doctrine/QueryBuilder/QueryBuilderGetQueryDynamicReturnTypeExtension.php (hunk 1 above)
Comment (janedbal) [reply src/Type/Doctrine/QueryBuilder/QueryBuilderGetQueryDynamicReturnTypeExtension.php:163]: It is. This is the main thing that enables the whole thing. Maybe you missed array_slice above.

Comment (janedbal) [reply src/Type/Doctrine/QueryBuilder/QueryBuilderGetQueryDynamicReturnTypeExtension.php:163]: The idea here is to drop second argument (which is typically an expression) and do "dummy update" `alias.field = alias.field` instead.

Comment (ondrejmirtes) [reply src/Type/Doctrine/QueryBuilder/QueryBuilderGetQueryDynamicReturnTypeExtension.php:163]: Oh alright, I know less and less about this extension every day :D

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**